uk_bin_collection/uk_bin_collection/councils/HuntingdonDistrictCouncil.py

```
@@ -10,49 +14,99 @@
-                        ).strftime(date_format),
-                    }
+                raise ValueError(
+                    f"Failed to parse bin type from text: '{text}'. "
+                    "The page format may have changed."
                 )
 
+            date_text = strong_tag.get_text(strip=True)
+            try:
+                collection_date = datetime.strptime(
+                    date_text, "%A %d %B %Y"
+                ).strftime(date_format)
```

Ruff (0.14.14)

[error] `date_format` may be undefined, or defined from star imports

(F405)